runtime/oti/VMHelpers.hpp

```diff
@@ -1402,16 +1402,20 @@ class VM_VMHelpers
 	static VMINLINE bool
 	threadIsInterruptedImpl(J9VMThread *currentThread, j9object_t threadObject)
 	{
+		J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, threadObject);
 		bool result = false;
+		
 		/* If the thread is alive, ask the OS thread.  Otherwise, answer false. */
-		if (J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject)) {
-			J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, threadObject);
-			if (NULL != targetThread) {
-				if (omrthread_interrupted(targetThread->osThread)) {
-					result = true;
-				}
+		if (NULL != targetThread) {
+			if (J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject)
+			&& omrthread_interrupted(targetThread->osThread)
+			) {
+				result = true;
 			}
+		} else {
+			result = J9VMJAVALANGTHREAD_DEADINTERRUPT(currentThread, threadObject);
```

Comment:
This also needs to occur when J9VMJAVALANGTHREAD_STARTED is false.

Reply:
Fixed as suggested above.